Scripts/Runtime/Util/Jobs/ParallelCollectionUtil.cs

```diff
@@ -92,12 +101,37 @@ public static int CollectionSizeForMaxThreads()
         /// <returns>The collection index to use</returns>
         public static int CollectionIndexForThread(int nativeThreadIndex)
         {
-            // JobsUtility.JobWorkerMaximumCount = 15
-            // index  0, 1, 2, 3, 4, 5, 6, 7, 8, 9,  10, 11, 12, 13, 14, 15
-            // thread 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 17
-            return (nativeThreadIndex <= JOB_WORKER_MAXIMUM_COUNT)
-                ? nativeThreadIndex - 1
-                : JOB_WORKER_MAXIMUM_COUNT;
+            DetectMultipleXThreads(nativeThreadIndex);
+            Debug.Assert(nativeThreadIndex > 0 && nativeThreadIndex <= JobsUtility.MaxJobThreadCount);
+            return math.min(nativeThreadIndex - 1, JOB_WORKER_MAXIMUM_COUNT.Data);
+        }
+
+
+        [Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
```

Comment:
Could probably just move this conditional onto the `ThreadHelper` class and its `DetectMultipleXThreads` method then call directly. I don't think this private method adds any value.

Reply:
I tried but Burst gets mad. 

We're not allowed to call a static method from inside a bursted function. So we call the private function which is BurstDiscarded which is then allowed to call the static function.

Reply:
Huh, maybe you'd have to bust discard the whole `ThreadHelper` class for it to work. In any case, given the restrictions, this method is fine.

Reply:
It seems it has to do with how Burst under the hood works. A static class gets created two times. Once for managed land and once for Burst. 

https://forum.unity.com/threads/burst-static-restrictions.536928/#post-4871255

Regardless, `BurstDiscard` only works on methods, not classes :(